enterprise/server/cmd/ci_runner/main.go

```diff
@@ -2272,3 +2285,50 @@ func runCredentialHelper() error {
 		return nil
 	}
 }
+
+// Attempts to free up disk space if necessary.
+func reclaimDiskSpace(ctx context.Context) error {
+	if runtime.GOOS != "linux" {
+		// Only do this on Linux VMs for now.
+		return nil
+	}
+	usage, err := diskUsageFraction()
+	if err != nil {
+		return fmt.Errorf("get disk usage: %w", err)
+	}
+	if usage < highDiskUsageThreshold {
+		return nil
+	}
+
+	// We should be in the git repo root at this point - run git gc.
+	log.Infof("Running git gc...")
+	if _, err := git(ctx, os.Stderr, "gc"); err != nil {
```

Comment:
Hmm, my concern is that with much garbage left over, git-gc could take a long time to complete. And this operation will block the PR check update, right?

I think we should default running these `git-maintenance` operations unconditionally instead:

```
git maintenance run loose-objects
git maintenance run incremental-repack
git maintenance run pack-refs
```

These commands are lighter weight and running them unconditionally each Invocation will divide out the total cleanup cost, avoiding 1 big blocking gc run (which, for larger repos, could cause OOM during packfile creation).

The caveat is that these are only available starting from Git 2.31.0

Reply:
> this operation will block the PR check update, right?

No, this cleanup is run after the workflow BES stream is closed, and the git statuses are published by the BES server.

> I think we should default running these git-maintenance operations unconditionally instead

These commands seem to behave better when running multiple times in a row, so running unconditionally SGTM (updated the PR)